extensions/postgres/non_transactional.go

```diff
@@ -72,3 +72,24 @@ func (d dbSession) BatchDeleteWorkerTask(
 	_, err := d.db.ExecContext(ctx, batchDeleteWorkerTaskQuery, filter.ShardId, filter.MinTaskSequenceInclusive, filter.MaxTaskSequenceInclusive)
 	return err
 }
+
+const batchSelectTimerTasksOfFirstPageQuery = `SELECT 
+    shard_id, fire_timestamp, task_sequence, process_execution_id, state_id, state_id_sequence, task_type, info
+	FROM xdb_sys_timer_tasks WHERE shard_id = $1 AND fire_timestamp>=$2 AND task_sequence>= $3 ORDER BY task_sequence ASC LIMIT $4`
+
+func (d dbSession) BatchSelectTimerTasks(ctx context.Context, filter extensions.TimerTaskRangeSelectFilter) ([]extensions.TimerTaskRow, error) {
```

Comment:
I am not sure whether this query can return the desired data. E.g., if the table contains:
```
task_sequence, fire_timestamp
1, 3
2, 2
3, 1
```
And limit is 2. Then we should get the following rows ordering by the fire_timestamp?
```
3, 1
2, 2
```